homeassistant/helpers/trigger.py

```diff
@@ -68,6 +69,48 @@ class TriggerInfo(TypedDict):
     trigger_data: TriggerData
 
 
+class PluggableAction:
+    """A pluggable action handler."""
+
+    def __init__(self, update: CALLBACK_TYPE | None = None) -> None:
+        """Initialize."""
+        self._update = update
+        self._actions: dict[
```

Comment:
I think there's a general problem with this approach of storing the trigger with the integration currently. If the integration is reloaded the previously stored triggers are lost as triggers are currently only attached when the user sets up the trigger or when reloading automations etc.

Until we have a solution for this, I'd be hesitant to creating a helper that more integrations can use.

Reply:
Aaargh!
So should https://github.com/home-assistant/core/pull/80393 be merged as it is?

Reply:
I think we should discuss what to do for these media players first. We normally don't want to fire custom events from entities. We now allow integrations to define their own triggers. But we didn't account for the problem of storing the triggers or re-attaching them when needed.